tests/subsys/event_manager/src/modules/test_multicontext.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+
+#include <test_events.h>
+#include <multicontext_event.h>
+
+#include "test_multicontext_config.h"
+
+#define MODULE test_multictx
+#define THREAD_STACK_SIZE 400
+
+
+static void send_event(int a, bool sleep)
+{
+	struct multicontext_event *ev = new_multicontext_event();
+
+	/* For every event both values should be the same -
+	 * used to check if Event Manager sends proper values.
+	 */
+	ev->val1 = a;
+	if (sleep) {
+		k_sleep(5);
+	}
+	ev->val2 = a;
+
+	EVENT_SUBMIT(ev);
+}
+
+static void timer_handler(struct k_timer *timer_id)
+{
+	send_event(SOURCE_ISR, false);
+}
+
+static K_TIMER_DEFINE(test_timer, timer_handler, NULL);
+static K_THREAD_STACK_DEFINE(thread_stack1, THREAD_STACK_SIZE);
+static K_THREAD_STACK_DEFINE(thread_stack2, THREAD_STACK_SIZE);
+
+static struct k_thread thread1;
+static struct k_thread thread2;
+static enum test_id cur_test_id;
+
+static void thread1_fn(void)
+{
+	send_event(SOURCE_T1, true);
+}
+
+static void thread2_fn(void)
+{
+	k_timer_start(&test_timer, K_MSEC(2), 0);
+	send_event(SOURCE_T2, true);
+}
+
+static void start_test(void)
+{
+	k_thread_create(&thread1, thread_stack1,
+			THREAD_STACK_SIZE,
+			(k_thread_entry_t)thread1_fn,
+			NULL, NULL, NULL,
+			K_PRIO_COOP(1), 0, K_NO_WAIT);
```

Comment:
You named it `THREAD1_PRIORITY` in header

Reply:
done